profiler/src/ProfilerEngine/Datadog.Profiler.Native/CorProfilerCallback.cpp

```diff
@@ -418,7 +427,8 @@ bool CorProfilerCallback::InitializeServices()
         _pCodeHotspotsThreadList,
         _pWallTimeProvider,
         _pCpuTimeProvider,
-        _metricsRegistry);
+        _metricsRegistry,
+        pool);
```

Comment:
If we want to track memory allocation per provider, we might need to "tag" the pool

Reply:
that's not part of this PR :P